# Report 1: Genome/transcriptome assembly and quality check

The task was done on the cluster. To access to the cluster the command **ssh agreshno@bea81** was used.

A directory **/nfs/scistore03/vicosgrp/Bioinformatics_2020/SASCHA_module1** was created with a command **mkdir**. Everything was done in that directory. All scripts were created and edited with the text editor Pico. To manage jobs in the cluster the program SLURM was used.

First, [test.sh](https://github.com/agreshno/bioinf/blob/master/test.sh) script was created. 

The first line of any script is the same: **#!/bin/bash**. 

Command | Meaning
-|-
#SBATCH --partition=defaultp | which queue to put it into 
#SBATCH --job-name=test | the name of the job
#SBATCH --output=test_output | the name of the output of the job
#SBATCH -c 5 | number of CPU cores to use within one node
#SBATCH --time=96:00:00 | max runtime
#SBATCH --mem=10G | the amount of RAM used by your job in GigaBytes
#SBATCH --mail-user=aleksandra.greshnova@ist.ac.at | -
#SBATCH --mail-type=ALL | to send emails when a job starts, it is finished or it exits
#SBATCH --no-requeue | do not requeue the job in the case it fails
#SBATCH --export=NONE | -
unset SLURM_EXPORT_ENV | do not export the local environment to the compute nodes
export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK | set the number of threads to the SLURM internal variable
srun echo "example text" | to print text

A command to check the status of the queue: **squeue -u agreshno**

A command to cancel a job: **scancel JOB_ID**

To use a reserved node the line 

**#SBATCH --partition=defaultp** 

was replaced with:

**#SBATCH --partition=bigmem**

**#SBATCH --reservation=vicores_2**

## Part I. Assembly

### 1. Sequencing reads

**Sample**: Human Fetal Brain RNA-Seq; Homo sapiens; RNA-Seq
SRA run ID: SRR4787052

The dataset was downloaded from SRA (Sequencing Read Archive) NCBI database with [download.sh](https://github.com/agreshno/bioinf/blob/master/download.sh) script through command line. The job was added to the queue with a command **sbatch download.sh**.

To download data the SRA-toolkit (version 2.8.1-3) was used.

Command | Meaning
-|-
**module load** SRA-Toolkit/2.8.1-3 | to upload the SRA-toolkit module
**srun** fastq-dump --skip-technical --readids --split-files SRR4787052 | to run a command
srun **fastq-dump** --skip-technical --readids --split-files SRR4787052 | to retrieve fastq files
srun fastq-dump **--skip-technical** --readids --split-files SRR4787052 | to keep only biological reads
srun fastq-dump --skip-technical **--readids** --split-files SRR4787052 | to keep original read IDs
srun fastq-dump --skip-technical --readids **--split-files** SRR4787052 | to create two files for paired-end reads

**Short description of the data:** Raw whole transcriptome data generated by RNA sequencing of 39 commonly-used neuroblastoma cell lines. 39 neuroblastoma cell lines with varying genomic characteristics, disease stages, and phases of therapy were sequenced. NextSeq 500 system by Illumina was used for sequencing.  

![basic_stat](https://github.com/agreshno/bioinf/blob/master/basic_stat.png)

Total number of reads was 63052537 & length of reads varied from 35 to 101 bp. The insert size (provided by the authors) was 200 bps. 

### 2. Reads quality and filtering

Assessment of the quality of the reads was performed was a diagnostic tool FastQC. [fastqc_test.sh](https://github.com/agreshno/bioinf/blob/master/fastqc_test.sh) script was created to perform the analysis.

![seq qual](https://github.com/agreshno/bioinf/blob/master/seq_qual.png)

Sequence quality was good even before any procedure. 

![per seq gc](https://github.com/agreshno/bioinf/blob/master/per_seq_gc.png)

One peak on this graph indicates that there was no contamination with RNA from other species.

![n cont](https://github.com/agreshno/bioinf/blob/master/n%20content.png)

N content was low which means that there was no gaps in the reads.

![adapt cont](https://github.com/agreshno/bioinf/blob/master/adapt%20cont.png)

Adapter content was low which means that all of the adapters were removed.

The program Trimmomatic was used to trim/remove low quality reads, or reads that have adapter sequences in them. 

[clean_reads.sh](https://github.com/agreshno/bioinf/blob/master/clean_reads.sh) script was used with the following commands:

Command | Meaning
-|-
module load java | download java module
java -jar | to install java
trimmomatic-0.36.jar | the actual java program
ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 | to remove adaptor sequences used by Illumina
LEADING:3 | to remove bad quality base-pairs at the beginning of the read
TRAILING:3 | to remove low quality base-pairs at the end of the read
SLIDINGWINDOW:4:15 | to run a sliding window through the read and remove sequences of low-quality base-pairs in the middle of the read
MINLEN:36 | to dump all reads that are smaller than 36 base-pairs after the trimming procedure

These are the most common parameters for RNA-Seq data. 

### 3. Transcriptome assembly

The program SOAPdenovo-trans was used to assemble the transcriptome. Two use this program two files were created:
1. [soapconfig.txt](https://github.com/agreshno/bioinf/blob/master/soapconfig.txt) - a configuration file that contains the information for SOAPdenovo-trans about the sequencing libraries
2. [SOAPdenovo-trans.sh](https://github.com/agreshno/bioinf/blob/master/SOAPdenovo-trans.sh) - a SLURM script

Command | Meaning
-|-
module load SOAPdenovoTrans | to download SOAPdenovo-trans program
srun SOAPdenovo-Trans-31mer all -s soapconfig.txt -K 31 -o TranscriptAssembly -p 16 1> TranscriptAssembly.stdout 2> TranscriptAssembly.stderr | to assemble transcriptome

